ts/features/idpay/initiative/configuration/xstate/provider.tsx

```diff
@@ -94,14 +93,12 @@ const IDPayConfigurationMachineProvider = (props: Props) => {
 
   const pmSessionManager = new SessionManager(getPaymentManagerSession);
 
-  const walletClient = createIDPayWalletClient(
+  const idPayClient = createIDPayClient(
```

Comment:
The problem here is that you are using a single client definition but still creating many clients for each flow.

Reply:
Yes that is true, we need to find an efficient way to provide a single client instance.